code/game/machinery/poolcontroller.dm

```diff
@@ -0,0 +1,137 @@
+/obj/machinery/poolcontroller
+	name = "Pool Controller"
+	desc = "A controller for the nearby pool."
+	icon = 'icons/obj/airlock_machines.dmi'
+	icon_state = "airlock_control_standby"
+	var/list/linkedturfs = list() //List contains all of the linked pool turfs to this controller, assignment happens on New()
+	var/temperature = "normal" //The temperature of the pool, starts off on normal, which has no effects.
+	var/srange = 5 //The range of the search for pool turfs, change this for bigger or smaller pools.
+	var/linkedmist = list() //Used to keep track of created mist
+	var/misted = 0 //Used to check for mist.
+
+/obj/machinery/poolcontroller/New() //This proc automatically happens on world start
+	for(var/turf/simulated/floor/beach/water/W in range(srange,src)) //Search for /turf/simulated/floor/beach/water in the range of var/srange
+		src.linkedturfs += W //Add found pool turfs to the central list.
+	..() //Changed to call parent as per MarkvA's recommendation
+
+/obj/machinery/poolcontroller/emag_act(user as mob) //Emag_act, this is called when it is hit with a cryptographic sequencer.
+	if(!emagged) //If it is not already emagged, emag it.
+		user << "\red You disable \the [src]'s temperature safeguards." //Inform the mob of what emagging does.
+		emagged = 1 //Set the emag var to true.
+
+/obj/machinery/poolcontroller/attackby(obj/item/P as obj, mob/user as mob, params) //Proc is called when a user hits the pool controller with something.
+
+	if(istype(P,/obj/item/device/multitool)) //If the mob hits the pool controller with a multitool, reset the emagged status
+		if(emagged) //Check the emag status
+			user << "\red You re-enable \the [src]'s temperature safeguards." //Inform the user that they have just fixed the safeguards.
+			emagged = 0 //Set the emagged var to false.
+			return
+		else
+			user << "\red Nothing happens." //If not emagged, don't do anything, and don't tell the user that it can be emagged.
+
+		return //Return, nothing else needs to be done.
+	else //If it's not a multitool, defer to /obj/machinery/proc/attackby
+		..()
+
+/obj/machinery/poolcontroller/attack_hand()
+	if(!Adjacent(usr))
+		usr << "You moved away."
+		return
+
+	if(emagged) //Emagging unlocks more (deadly) options.
+		var/temp = input("What temperature would you like to set the pool to?", "Pool Temperature") in list("Hot","Cold", "Warm", "Cool", "Normal","Cancel") //Allow user to choose which temperature they want.
+
+		switch(temp) //Used for setting the actual temperature var based on user input.
+			if("Hot")
+				miston()
+				temperature = "hot"
+				usr << "<span class='warning'>You flick the pool temperature to [temperature](WARNING).</span>" //Differ from standard message to make sure the user understands the temperature is harmful.
+				return
+			if("Cold")
```

Comment:
recommend a different descriptors for these; "scalding" and "frigid" to get across how insanely hot/cold they are.